main.py

```diff
@@ -148,62 +203,125 @@ def file_paths_to_dict(resume_file: Path | None, plain_text_resume_file: Path) -
+        bot.set_secrets(email, password)
+        logger.info("Secrets set successfully.")
+
+        logger.info("Initializing login component...")
+        login_component = AIHawkAuthenticator(driver=browser, bot_facade=bot)
+        logger.debug(f"Login component created: {login_component}")
+
+        bot.login_component = login_component
+        logger.debug("Login component set in bot facade.")
+
+        logger.info("Initializing GPT Answerer component...")
         gpt_answerer_component = GPTAnswerer(parameters, llm_api_key)
-        bot = AIHawkBotFacade(login_component, apply_component)
+        logger.debug(f"GPT Answerer component created: {gpt_answerer_component}")
+
+
+        logger.info("Setting job application profile and resume...")
         bot.set_job_application_profile_and_resume(job_application_profile_object, resume_object)
+        logger.info("Job application profile and resume set successfully.")
+
+        logger.info("Setting GPT Answerer and resume generator...")
         bot.set_gpt_answerer_and_resume_generator(gpt_answerer_component, resume_generator_manager)
+        logger.info("GPT Answerer and resume generator set successfully.")
+
+        logger.info("Setting additional parameters for the bot...")
         bot.set_parameters(parameters)
+        logger.info("Parameters set successfully.")
+
+        logger.info("Starting bot login process...")
         bot.start_login()
+        logger.info("Login process completed successfully.")
+
+        logger.info("Starting job application process...")
         bot.start_apply()
+        logger.info("Job application process completed successfully.")
+
     except WebDriverException as e:
         logger.error(f"WebDriver error occurred: {e}")
     except Exception as e:
+        logger.error("An unexpected error occurred in create_and_run_bot:")
+        logger.exception(e)
         raise RuntimeError(f"Error running the bot: {str(e)}")
 
 
+
 @click.command()
-@click.option('--resume', type=click.Path(exists=True, file_okay=True, dir_okay=False, path_type=Path), help="Path to the resume PDF file")
+@click.option('--resume', type=click.Path(exists=True, file_okay=True, dir_okay=False, path_type=Path),
+              help="Path to the resume PDF file")
 def main(resume: Path = None):
     try:
         data_folder = Path("data_folder")
         secrets_file, config_file, plain_text_resume_file, output_folder = FileManager.validate_data_folder(data_folder)
 
         parameters = ConfigValidator.validate_config(config_file)
-        llm_api_key = ConfigValidator.validate_secrets(secrets_file)
+        email, password, llm_api_key = ConfigValidator.validate_secrets(secrets_file)
 
         parameters['uploads'] = FileManager.file_paths_to_dict(resume, plain_text_resume_file)
         parameters['outputFileDirectory'] = output_folder
 
-        create_and_run_bot(parameters, llm_api_key)
+        create_and_run_bot(email, password, parameters, llm_api_key)
```

Comment:
main has too many unnecessary logs, and its making code unreadable. The raise error is sufficient in all the case.

Reply:
Can we use wrapper if we need to keep so many debug logs?
```
class LoggerWrapper:
    def __init__(self, logger):
        self.logger = logger
        self.debug_mode = os.environ.get('DEBUG', 'False').lower() == 'true'

    def debug(self, message, *args, **kwargs):
        if self.debug_mode:
            self.logger.debug(message, *args, **kwargs)

    def info(self, message, *args, **kwargs):
        self.logger.info(message, *args, **kwargs)

    def __getattr__(self, name):
        return getattr(self.logger, name)
```